dev/validate_bom__config.py

```diff
@@ -936,38 +941,54 @@ def init_argument_parser(self, parser, defaults):
 
   def validate_options(self, options):
     """Implements interface."""
-    check_options_set(options, ['pubsub_google_project', 'pubsub_google_credentials_path', 'pubsub_google_subscription_name', 'pubsub_google_template_path'])
-    check_path_exists(options.pubsub_google_credentials_path, "pubsub_credentials_path")
-    check_path_exists(options.pubsub_google_template_path, "pubsub_template_path")
-    options.pubsub_google_enabled = options.pubsub_google_subscription_name is not None
+    options.pubsub_google_enabled = options.pubsub_google_project is not None
+
+    if not options.pubsub_google_enabled:
+      return
+
+    check_options_set(
+        options,
+        ['pubsub_google_project', 'pubsub_google_credentials_path',
+         'pubsub_google_subscription_name', 'pubsub_google_template_path'])
+    check_path_exists(options.pubsub_google_credentials_path,
+                      "pubsub_credentials_path")
+    check_path_exists(options.pubsub_google_template_path,
+                      "pubsub_template_path")
 
   def add_config(self, options, script):
     """Implements interface."""
-    if options.pubsub_google_name is None:
+    if not options.pubsub_google_enabled:
       return
 
     script.append('hal -q --log=info config pubsub google enable')
     subscription_cmd = ['hal -q --log=info config pubsub google subscription']
     if options.pubsub_google_name:
       subscription_cmd.append('add {}'.format(options.pubsub_google_name))
     if options.pubsub_google_project:
-      subscription_cmd.append('--project {}'.format(options.pubsub_google_project))
+      subscription_cmd.append('--project ' + options.pubsub_google_project)
```

Comment:
On a second look at this, I think all of these are required, so if any are missing any then we should fail. Does `check_options_set` fail if any of the arguments are `None`? If so I think we can assume without guarding each of these with an `if`.

Reply:
Discussed out of band, LGTM.